src/backend/cuda/pup/yaksuri_cudai_pup.c

```diff
@@ -9,6 +9,42 @@
 #include "yaksi.h"
 #include "yaksuri_cudai.h"
 
+#define THREAD_BLOCK_SIZE  (256)
+#define MAX_GRIDSZ_X       ((1ULL << 31) - 1)
+#define MAX_GRIDSZ_Y       (65535)
+#define MAX_GRIDSZ_Z       (65535)
+
+static int get_thread_block_dims(uint64_t count, yaksi_type_s * type, int *n_threads,
+                                 int *n_blocks_x, int *n_blocks_y, int *n_blocks_z)
+{
+    int rc = YAKSA_SUCCESS;
+    yaksuri_cudai_type_s *cuda_type = (yaksuri_cudai_type_s *) type->backend.cuda.priv;
+
+    *n_threads = THREAD_BLOCK_SIZE;
+    uint64_t n_blocks = count * cuda_type->num_elements / THREAD_BLOCK_SIZE;
+    n_blocks += ! !(count * cuda_type->num_elements % THREAD_BLOCK_SIZE);
+
```

Comment:
For correctness, should this return an error code if the number of blocks exceeds the max allowed size? Or simply assert?

Reply:
Oh I think I have commented too late

Reply:
That would be more than the size of `int64_t`.  At that point, we'd need to change a whole lot of code in yaksa to make it work, and an assert would not be sufficient.